### PR TITLE
STFE-9: implementation of teacher registration form

### PR DESCRIPTION
**Pull request by mohammed-sharraf-sazim**
## Description of Change

- Addition of Teacher Registration form

## Associated Ticket Link(s)
- https://trello.com/c/2EMl9xBY/9-subtask-21-design-the-teacher-registration-form

## Screenshots / Screen Recordings
- Screen Recording:

https://github.com/user-attachments/assets/43814dac-d729-4d8e-a172-535270adf468

- Screenshot(Large Screen):
Screenshot 2024-08-27 at 9 10 38 PM
- Screenshot(Medium Screen):
Screenshot 2024-08-27 at 9 11 34 PM
- Screenshot(Small Screen):
Screenshot 2024-08-27 at 9 10 38 PM

### CODE REVIEW

**Comment by mohammed-sharraf-sazim** [comment]
Sorry the screenshot for small screen will be this:
Screenshot 2024-08-27 at 9 12 15 PM

File: modules/RegisterUser/components/TeacherRegistration/TeacherForm.tsx [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+  });
+  const options = [
+    { label: "Maths", value: "Maths" },
+    { label: "Physics", value: "Physics" },
+    { label: "Chemistry", value: "Chemistry" },
+    { label: "Biology", value: "Biology" },
+    { label: "English", value: "English" },
+    { label: "Bangla", value: "Bangla" },
+    { label: "Computer Science", value: "Computer Science" },
+  ];
```
Comment (nur-zaman-sazim) [inline modules/RegisterUser/components/TeacherRegistration/TeacherForm.tsx:38]: Please move this to constants file

**Review by nur-zaman-sazim — CHANGES_REQUESTED** [review]

File: modules/RegisterUser/components/TeacherRegistration/TeacherForm.tsx [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+} from "@/shared/components/shadui/select";
+import { ETeacherHighestEducation } from "@/shared/typedefs";
+
+import { teacherRegistrationSchemaResolver } from "./TeacherForm.schema";
+
+const TeacherForm = () => {
+  const form = useForm({
+    resolver: teacherRegistrationSchemaResolver,
+  });
+  const options = [
```
Comment (mahir-mohsin-sazim) [inline modules/RegisterUser/components/TeacherRegistration/TeacherForm.tsx:30]: Have this in a constants.ts file

File: modules/RegisterUser/containers/TeacherRegistrationContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+import TeacherForm from "../components/TeacherRegistration/TeacherForm";
```
Comment (mahir-mohsin-sazim) [inline modules/RegisterUser/containers/TeacherRegistrationContainer.tsx:3]: Use named exports for your own components

File: styles/globals.css [hunk 4]
```diff
@@ -5,72 +5,65 @@
...
+    --chart-5: 340 75% 55%;
   }
 }
 
 @layer base {
   * {
     @apply border-border;
   }
-  body {
+  /* body {
```
Comment (mahir-mohsin-sazim) [inline styles/globals.css:66]: Remove this block if you do not require it

**Review by mahir-mohsin-sazim — CHANGES_REQUESTED** [review]

File: modules/RegisterUser/components/TeacherRegistration/TeacherForm.tsx (hunk 1 above)
Comment (mohammed-sharraf-sazim) [reply modules/RegisterUser/components/TeacherRegistration/TeacherForm.tsx]: Moved to the constants

File: modules/RegisterUser/components/TeacherRegistration/TeacherForm.tsx (hunk 2 above)
Comment (mohammed-sharraf-sazim) [reply modules/RegisterUser/components/TeacherRegistration/TeacherForm.tsx]: Moved

File: modules/RegisterUser/containers/TeacherRegistrationContainer.tsx (hunk 3 above)
Comment (mohammed-sharraf-sazim) [reply modules/RegisterUser/containers/TeacherRegistrationContainer.tsx:3]: Used named exports

File: styles/globals.css (hunk 4 above)
Comment (mohammed-sharraf-sazim) [reply styles/globals.css]: Removed

**Review by nur-zaman-sazim — APPROVED** [review]

**Review by nur-zaman-sazim — APPROVED** [review]

**Review by mahir-mohsin-sazim — APPROVED** [review]

**Pull request merged**